lib/Runtime/Types/TypePropertyCache.cpp

```diff
@@ -296,8 +299,10 @@ namespace Js
                 requestContext);
             return true;
         }
-
-        *propertyValue = CrossSite::MarshalVar(requestContext, *propertyValue);
+        else
+        {
+            *propertyValue = CrossSite::MarshalVar(requestContext, *propertyValue);
```

Comment:
Do we need to marshal here? We already had this check before and didn't marshal in the false case..

Reply:
code reaches here only when `! if(prototypeObjectWithProperty->GetScriptContext() == requestContext)`

Reply:
Right, but it already did before without marshalling. Is there an explicit need now?

Reply:
Not sure. I just added `else` there.

Reply:
Oh sorry the diff wasn't showing properly. That's fine.